File: memcached/golang/routes/route.go
package routes

import (
	"note-golang-memcached/controllers"
	"note-golang-memcached/services"
	"note-golang-memcached/utils"

	"github.com/labstack/echo/v4"
)

func SetMemcachedRoute(e *echo.Echo, memcachedUtil utils.MemcachedUtil) {
	memcachedService := services.NewMemcachedService(memcachedUtil)
	memcachedController := controllers.NewMemcachedController(memcachedService)
	e.POST("/api/v1/memcached", memcachedController.Set)
	e.GET("/api/v1/memcached/:id", memcachedController.Get)
	e.DELETE("/api/v1/memcached", memcachedController.Delete)
}

File: memcached/rust/src/routes/route.rs
use std::sync::Arc;

use axum::{routing::{delete, get, post}, Router};

use crate::{controllers::memcached_controller::{delete_handler, flush_handler, get_handler, set_handler}, services::memcached_service::MemcachedServiceImpl, states::app_state::AppState, utils::memcached_util::MemcachedUtilImpl};

pub fn set_route(memcached_util: Arc<MemcachedUtilImpl>) -> Router {
    let memcached_service = Arc::new(MemcachedServiceImpl::new(memcached_util));
    let state = Arc::new(AppState{memcached_service: memcached_service.clone()});
    Router::new()
        .route("/memcached", post(set_handler))
        .route("/memcached/{id}", get(get_handler))
        .route("/memcached", delete(delete_handler))
        .route("/memcached/flush", post(flush_handler))
        .with_state(state)
}